server/src/modules/rules/getter/plex-getter.service.ts

```diff
@@ -119,17 +119,14 @@ export class PlexGetterService {
                   return null;
                 });
 
-              var i = allViewers.length;
-              while (i--) {
+              allViewers.forEach((el, idx) => {
```

Comment:
When I used `forEach` like this and tested it, the result was that it did not worked for seasons if there was only 1 or 2 episodes and about 5 users on the Plex server. Not all users were examined because the `length` of elements and with that the `index` changed with the `allViewers.splice()` function and therefor not all users are examined on the first episode in the for loop. it took 3-4 episodes to have all users examined. The only solution I found was looping the index counting backwards if you want to remove elements while looping.

Reply:
https://stackoverflow.com/a/9882349

Reply:
Thanks for this! I wasn't aware of this behavior with forEach & splice.
Did you already try something like this? If I'm not mistaken, it also mitigates the issue.

    const arrLength = allViewers.length - 1;
    allViewers
      .slice()
      .reverse()
      .forEach((el, idx) => {
        if (
          !viewers ||
          !viewers.find((viewEl) => el.plexId === viewEl.accountID)
        ) {
          allViewers.splice(arrLength - idx, 1);
        }
      });

Reply:
Let's continue this conversation in https://github.com/jorenn92/Maintainerr/pull/449. If you notice an issue with this approach, please let me know.